magicctapipe/io/tests/test_io.py

```diff
@@ -19,619 +21,721 @@
+            assert np.all(data["intensity"] > 50)
+
+    def test_load_train_data_files_p(self, p_stereo):
+        """
+        Check dictionary
+        """
+
+        events = load_train_data_files(str(p_stereo[0]))
+        assert list(events.keys()) == ["M1_M2", "LST1_M1", "LST1_M2", "LST1_M1_M2"]
+        data = events["LST1_M1"]
+        assert np.all(data["combo_type"] == 1)
+        assert "off_axis" in data.columns
+        assert "true_event_class" not in data.columns
+
+    def test_load_train_data_files_g(self, gamma_stereo):
+        """
+        Check dictionary
+        """
+
+        events = load_train_data_files(str(gamma_stereo[0]))
+        assert list(events.keys()) == ["M1_M2", "LST1_M1", "LST1_M2", "LST1_M1_M2"]
+        data = events["LST1_M1"]
+        assert np.all(data["combo_type"] == 1)
+        assert "off_axis" in data.columns
+        assert "true_event_class" not in data.columns
+
+    def test_load_train_data_files_off(self, gamma_stereo):
+        """
+        Check off-axis cut
+        """
+        events = load_train_data_files(
+            str(gamma_stereo[0]), offaxis_min="0.2 deg", offaxis_max="0.5 deg"
+        )
+        data = events["LST1_M1"]
+        assert np.all(data["off_axis"] >= 0.2)
+        assert np.all(data["off_axis"] <= 0.5)
 
+    def test_load_train_data_files_exc(self, temp_train_exc):
+        """
+        Check on exceptions
+        """
 
-def test_load_mc_dl2_data_file_exc(p_dl2, gamma_dl2):
-    """
-    Check on event_type exceptions
-    """
-    dl2_mc = [p for p in gamma_dl2.glob("*")] + [p for p in p_dl2.glob("*")]
-    for file in dl2_mc:
-        event_type = "abc"
         with pytest.raises(
-            ValueError,
-            match=f"Unknown event type '{event_type}'.",
+            FileNotFoundError,
+            match="Could not find any DL1-stereo data files in the input directory.",
         ):
-            _, _, _ = load_mc_dl2_data_file(str(file), "width>0", event_type, "simple")
-
-
-def test_get_dl2_mean_mc(p_dl2, gamma_dl2):
-    """
-    Check on MC DL2
-    """
-    dl2_mc = [p for p in gamma_dl2.glob("*")] + [p for p in p_dl2.glob("*")]
-    for file in dl2_mc:
-        event_data = pd.read_hdf(str(file), key="events/parameters")
+            _ = load_train_data_files(str(temp_train_exc))
+
+    def test_load_train_data_files_tel_p(self, p_stereo, config_gen):
+        """
+        Check dictionary
+        """
+
+        events = load_train_data_files_tel(str(p_stereo[0]), config_gen)
+        assert list(events.keys()) == [1, 2, 3]
+        data = events[2]
+        assert "off_axis" in data.columns
+        assert "true_event_class" not in data.columns
+
+    def test_load_train_data_files_tel_g(self, gamma_stereo, config_gen):
+        """
+        Check dictionary
+        """
+
+        events = load_train_data_files_tel(str(gamma_stereo[0]), config_gen)
+        assert list(events.keys()) == [1, 2, 3]
+        data = events[1]
```

Comment:
any reason why you use second telescope for protons in the function above, and first for gammas here? If not and both can use the same you can again make just one function and run it for two parameters as I explained a few comments above